src/main/java/org/opentripplanner/routing/algorithm/filterchain/ItineraryListFilterChainBuilder.java

```diff
@@ -249,6 +257,8 @@ public ItineraryListFilterChain build() {
       filters.add(new AccessibilityScoreFilter(wheelchairMaxSlope));
     }
 
+    Optional.ofNullable(faresService).map(service -> filters.add(new FaresFilter(faresService)));
```

Comment:
A traditional if clause miht be more clear here, as it is used for the other filters

Reply:
Good point - I sometimes overdo it with the functional paradigms.